PennyMe/ViewController.swift

```diff
@@ -331,6 +345,13 @@ class ViewController: UIViewController, UITextFieldDelegate, UIGestureRecognizer
         }
     }
     
+    
+    func checkMachineShouldBeVisible(status: String, machineStatus: String) -> Bool {
+        //checks based on the status and the current user defaults whether a machine should be visible
+        // should be shown if available and status toggle is on, or if not available and retired toggle is on
+        return ((includedStates.contains(status) && (machineStatus == "available")) || (includedStates.contains("retired") && (machineStatus != "available")))
```

Comment:
same logic as before, now moved into a method to be used in `addAnnotationsIteratively` and in `check_json_dict`